packages/plans-grid-next/src/types.ts

```diff
@@ -197,11 +197,37 @@ export type GridContextProps = {
 	enableFeatureTooltips?: boolean;
 	featureGroupMap: Partial< FeatureGroupMap >;
 	hideUnsupportedFeatures?: boolean;
+	enableCategorisedFeatures?: boolean;
+
 	/**
-	 * `enableCategorisedFeatures` is no longer exact, and probably best to rename.
-	 * It is only used for showing "Everything in [previous] plus".
+	 * Display the feature titles with a slightly larger font size
 	 */
-	enableCategorisedFeatures?: boolean;
+	enableLargeFeatureTitles?: boolean;
+
+	/**
+	 * Display the plan storage limit as a badge like "50GB" or as plain text like "50GB storage"
+	 */
+	enableStorageAsBadge?: boolean;
+
+	/**
+	 * Display the "Everything in [previous-plan], plus:" title above the features list
+	 */
+	enablePreviousFeaturesIncludedTitle?: boolean;
+
+	/**
+	 * Reduce the vertical spacing between each feature group
+	 */
+	enableReducedFeatureGroupSpacing?: boolean;
```

Comment:
I don't like how these `enableFoo`, `enableBar` (etc) props attempt to pattern-ize the idea of adding granular props to control experiment logic. Almost always this is the result of lacking a better underlying design. I don't want to see `enableX` being projected as an acceptable pattern that can span into some Nth dimension (anyone adding anything without due consideration). The vision with plans-grid package is to reduce the need for new/standalone props whenever we want to run an experiment, unless absolutely necessary.

For this case, there could be other options to consider:

- introduce optional render prop that consuming end can tailor design. This can stay, irrespective of experiment result
- overwrite the styling at consuming end (plans-features-main) and follow up to deploy the final change as the default grid styling once/if experiment completes successfully
- more?

Have we considered any of the other options?

Reply:
> introduce optional render props that consuming end can tailor design. This can stay, irrespective of experiment result. Can literally call them renderProps (interface RenderProps { Feature: ..., etc. })

Just to clarify are you referring to the render props pattern? ie.

```
<FeaturesGrid
  renderFeatureTitles={ () => ... }
  ...
/>
```

Or `renderProps` as a single prop controlling multiple display properties? ie.

```
<FeaturesGrid 
  renderProps={{
    largeFeatureTitles: true,
    ...
  }}
/>
```

Reply:
I am referring to the first/pattern, but let's also not make it super inviting to the "granularity effect". Meaning, it should be a holder of general components (as possible). I'd go for a type that looks like:

```
interface Renderers {
    Feature?: React.FunctionComponent< FeatureRendererProps >
}
```

and add that as an optional prop (`renderers`) on the common grid props and context.

p.s. Totally fine to handle this in a follow-up if it's anything more complex than it looks. Not blocking. We can create an issue and follow up on it with some priority.

Reply:
Ok let's tackle this change in a followup. It does sound relatively straightforward, however it would be good to have some time to explore in more detail (outside of this experiment). I've created an issue: https://github.com/Automattic/wp-calypso/issues/94458